### ISSUE TITLE
Treeview Tooltips circular structure error 

### ISSUE THREAD

**Issue by OGordon100** [CNFeffery/feffery-antd-components#45]
Following #41, I have found 2 bugs:

1) If setting tooltipProps{"title": "text"} instead of {'title':'text'}, Dash shows a circular error

2) If updating the treeData data directly, editing tooltipProps results in the same circular error (other properties can be updated dynamically without issue)

`TypeError: Converting circular structure to JSON

    --> starting at object with constructor 'Array'

    |     index 0 -> object with constructor 'Object'

    |     property 'title' -> object with constructor 'Object'

    |     ...

    |     property 'props' -> object with constructor 'Object'

    --- property 'treeData' closes the circle

    at JSON.stringify (<anonymous>)

    at handleServerside (http://127.0.0.1:8050/_dash-component-suites/dash/dash-renderer/build/dash_renderer.v2_6_2m1664201321.dev.js:795:19)

    at _callee$ (http://127.0.0.1:8050/_dash-component-suites/dash/dash-renderer/build/dash_renderer.v2_6_2m1664201321.dev.js:1169:24)

    at tryCatch (http://127.0.0.1:8050/_dash-component-suites/dash/dash-renderer/build/dash_renderer.v2_6_2m1664201321.dev.js:474:2404)

    at Generator._invoke (http://127.0.0.1:8050/_dash-component-suites/dash/dash-renderer/build/dash_renderer.v2_6_2m1664201321.dev.js:474:1964)

    at Generator.next (http://127.0.0.1:8050/_dash-component-suites/dash/dash-renderer/build/dash_renderer.v2_6_2m1664201321.dev.js:474:3255)

    at asyncGeneratorStep (http://127.0.0.1:8050/_dash-component-suites/dash/dash-renderer/build/dash_renderer.v2_6_2m1664201321.dev.js:482:103)

    at _next (http://127.0.0.1:8050/_dash-component-suites/dash/dash-renderer/build/dash_renderer.v2_6_2m1664201321.dev.js:484:194)

    at http://127.0.0.1:8050/_dash-component-suites/dash/dash-renderer/build/dash_renderer.v2_6_2m1664201321.dev.js:484:364

    at new Promise (<anonymous>)`

**Comment by CNFeffery** [comment]
@OGordon100 Can you provide a simple demo to reproduct these bug🧐.

**Comment by OGordon100** [comment]
@CNFeffery 

See the following sample code. The tree itself is dynamically created. 

Pressing "generate tree" and then "replace tooltip" will break the page. This does not crash when removing the tooltipProps entry. 

In my current code, I have the replace_tooltip function edit things like title, children, selected, etc. All this works fine, but raises the circularimport error if tooltipProps are specified.

`
import feffery_antd_components as fac
import dash_bootstrap_components as dbc
from dash_extensions.enrich import DashProxy, MultiplexerTransform, CycleBreakerTransform, html
from dash import Dash, Input, Output, State

app = DashProxy(__name__, prevent_initial_callbacks=True, suppress_callback_exceptions=True,
                transforms=[MultiplexerTransform(), CycleBreakerTransform()])


@app.callback(Output("tree", "children"),
              Input("make-button", "n_clicks"))
def make_tree(_):
    return tree


@app.callback(Output("tree", "children"),
              State("tree", "children"),
              Input("breaking-button", "n_clicks"))
def replace_tooltip(tree, _):
    return tree


tree = fac.AntdTree(id="tree-classifications", treeData=[
    {'title': 'a', 'key': '1', 'tooltipProps': {'title': 'test', 'placement': 'right'}, 'children': [
        {'title': 'b', 'key': '2', 'tooltipProps': {'title': 'test', 'placement': 'right'},
         'children': [{'title': 'c', 'key': '3', 'tooltipProps': {'title': 'test', 'placement': 'right'}},
                      {'title': 'd', 'key': '4', 'tooltipProps': {'title': 'test', 'placement': 'right'}}]},
        {'title': 'e', 'key': '5', 'tooltipProps': {'title': 'test', 'placement': 'right'},
         'children': [{'title': 'f', 'key': '6', 'tooltipProps': {'title': 'test', 'placement': 'right'}}]}]}],
                    checkable=True, draggable=False, selectable=False, checkStrictly=True, defaultExpandAll=True,
                    multiple=True)

generate_button = dbc.Button("Generate Tree", id="make-button")
breaking_button = dbc.Button("Replace Tooltip", id="breaking-button")

app.layout = html.Div([html.Div(id="tree"), generate_button, breaking_button])

if __name__ == '__main__':
    app.run_server(debug=True)
`

**Comment by CNFeffery** [comment]
@OGordon100 Fixed in `0.2.0rc15`, follow this demo:
![demo](https://user-images.githubusercontent.com/49147660/202348627-d1e219a4-46eb-43f2-b4ce-3b59e9ae5ffa.gif)

```Python
import dash
from dash import html
from datetime import datetime
import feffery_antd_components as fac
from dash.dependencies import Input, Output, State

app = dash.Dash(
    __name__,
    suppress_callback_exceptions=True
)

tree = fac.AntdTree(
    id='tree',
    defaultExpandAll=True,
    treeData=[
        {
            'title': '重庆市',
            'key': '重庆市',
            'tooltipProps': {
                'title': '😀tooltip'
            },
            'children': [
                {
                    'title': '渝北区',
                    'key': '渝北区',
                    'tooltipProps': {
                        'title': '🥳tooltip',
                        'placement': 'right'
                    }
                },
                {
                    'title': '江北区',
                    'key': '江北区',
                    'disabled': True
                }
            ]
        },
        {
            'title': '北京市',
            'key': '北京市',
            'children': [
                {
                    'title': '西城区',
                    'key': '西城区'
                },
                {
                    'title': '东城区',
                    'key': '东城区'
                }
            ]
        },
        {
            'title': '四川省',
            'key': '四川省',
            'children': [
                {
                    'title': '成都市',
                    'key': '成都市',
                    'children': [
                        {
                            'title': '天府新区',
                            'key': '天府新区'
                        }
                    ]
                }
            ]
        }
    ]
)

app.layout = html.Div(
    [
        fac.AntdSpace(
            [
                fac.AntdButton(
                    'generate tree',
                    id='generate-tree',
                    type='primary'
                ),
                fac.AntdButton(
                    'update tree tooltips',
                    id='update-tree-tooltips',
                    type='primary'
                )
            ]
        ),

        html.Div(
            id='tree-container'
        )
    ],
    style={
        'padding': '100px'
    }
)


@app.callback(
    Output('tree-container', 'children'),
    Input('generate-tree', 'nClicks'),
    prevent_initial_call=True
)
def generate_tree_demo(nClicks):

    return tree


@app.callback(
    Output('tree', 'treeData'),
    Input('update-tree-tooltips', 'nClicks'),
    State('tree', 'treeData'),
    prevent_initial_call=True
)
def update_tree_tooltips_demo(nClicks, treeData):

    treeData[0]['children'][0]['tooltipProps'] = {
        'title': '🥳tooltip ' + datetime.now().strftime('%Y-%m-%d %H:%M:%S'),
        'placement': 'right'
    }

    return treeData

if __name__ == '__main__':
    app.run(debug=True)

```

**Comment by OGordon100** [comment]
🎉🎉🎉

Thank you!
